src/outputgeneration/ParseTreeMapWriter.js

```diff
@@ -90,39 +95,34 @@ export class ParseTreeMapWriter extends ParseTreeWriter {
   }
 
   enterBranch(location) {
-    this.originate(location.start);
+    let {line, column, source} = location.start;
+    this.originate(line, column, source);
   }
 
   exitBranch(location) {
-    let position = location.end;
-    let endOfPreviousToken = {
-      line: position.line,
-      column: position.column ? position.column - 1 : 0,
-      source : position.source,
-    }
-    this.originate(endOfPreviousToken);
+    let {line, column, source} = location.end;
+    this.originate(line, column ? column - 1 : 0, source);
   }
 
   /**
   * Set the original coordinates for the generated position.
-  * @param {Object} position Traceur SourcePosition, line and col zero based.
+  * @param {number} line
+  * @param {number} column
+  * @param {SourceFile} source
   */
-  originate(position) {
-    let line = position.line + 1;  // source map lib uses one-based lines.
+  originate(line, column, source) {
```

Comment:
the type of the argument to this function was different based on the call sites.